src/somef/extract_software_type.py

```diff
@@ -0,0 +1,309 @@
+                                        constants.PROP_VALUE: 'https://w3id.org/software-types#Commandline_Application',
+                                        constants.PROP_TYPE: constants.URL,
+                                        
+                                    },
+                                    0.82,
+                                    constants.TECHNIQUE_HEURISTICS)
+    else:
+        metadata_result.add_result(constants.TYPE,
+                                    {
+                                        constants.PROP_VALUE: 'https://www.britannica.com/technology/software',
+                                        constants.PROP_TYPE: constants.URL,
+                                        
+                                    },
+                                    1,
+                                    constants.TECHNIQUE_HEURISTICS)
+    return metadata_result
+
+
+def check_notebooks(path_repo):
+    """Function which checks if the specified repository is a Notebook Application
+       depending on the extensions present and number of notebooks which contain code"""
+    total_notebooks = 0
+    code_notebooks = 0
+    total_files=0
+    script_extensions = ('.py','.exe','.sh', '.bat', '.cmd', '.jar', '.cpp', '.c', '.h', '.hpp', '.java', '.cs', '.vb', '.php', '.pl', '.rb')
+    
+
+    bad_extensions=False
+    for root, dirs, files in os.walk(path_repo):
+        for file in files:
+            if file.endswith(media_extensions):
+                continue
+            else:
+                total_files+=1
+            if file.endswith(script_extensions):
+                bad_extensions=True
+
+            if file.endswith((".ipynb", ".rmd",'.Rmd',".jl")):
+                notebook_path = os.path.join(root, file)
+                try:
+                    has_code = False
+                    num_code_cells = 0
+                    num_total_cells = 0
+
+                    if file.endswith(".ipynb"):
+                        nb = nbformat.read(notebook_path, as_version=4)
+                        for cell in nb['cells']:
+                            if cell['cell_type'] == 'code':
+                                num_total_cells += 1
+                                if cell['source'].strip():
+                                    num_code_cells += 1
+                                    has_code = True
+                        if has_code and num_code_cells / num_total_cells >= 0.55:
+                            code_notebooks += 1
+                    elif file.endswith(".rmd") or file.endswith('.Rmd'):
+                        code_notebooks += 1
+                    elif file.endswith('.jl'):
+                        code_notebooks +=1
+
+
+                    total_notebooks += 1
+
+                except Exception as e:
+                    print(f"Error reading notebook file {notebook_path}: {str(e)}")
+                    pass
+
+    if code_notebooks/total_notebooks>=0.5 and bad_extensions==False:
```

Comment:
This value is a little arbitrary. Please explain, and put in a variable that is documented somewhere.